doc/source/contributing.rst

```diff
@@ -4,13 +4,574 @@
+to see if the issue has already been reported and/or fixed.
+
+Bug reports must:
+
+#. Include a short, self-contained Python snippet reproducing the problem.
+   You can have the code formatted nicely by using `GitHub Flavored Markdown
+   <http://github.github.com/github-flavored-markdown/>`_: ::
+
+      ```python
+      >>> from pandas import DataFrame
+      >>> df = DataFrame(...)
+      ...
+      ```
+
+#. Include the full version string of *pandas* and its dependencies. In recent (>0.12) versions
+   of *pandas* you can use a built in function: ::
+
+      >>> from pandas.util.print_versions import show_versions
+      >>> show_versions()
+
+   and in 0.13.1 onwards: ::
+
+      >>> pd.show_versions()
+   
+#. Explain why the current behavior is wrong/not desired and what you expect instead.
+
+The issue will then show up to the *pandas* community and be open to comments/ideas from others.
+
+Working with the code
+=====================
+
+Now that you have an issue you want to fix, enhancement to add, or documentation to improve, 
+you need to learn how to work with GitHub and the *pandas* code base.
+
+Version Control, Git, and GitHub
+--------------------------------
+
+To the new user, working with Git is one of the more daunting aspects of contributing to *pandas*.  
+It can very quickly become overwhelming, but sticking to the guidelines below will make the process 
+straightforward and will work without much trouble.  As always, if you are having difficulties please 
+feel free to ask for help.
+
+The code is hosted on `GitHub <https://www.github.com/pydata/pandas>`_. To
+contribute you will need to sign up for a `free GitHub account
+<https://github.com/signup/free>`_. We use `Git <http://git-scm.com/>`_ for
+version control to allow many people to work together on the project.
+
+Some great resources for learning git:
+
+ * the `GitHub help pages <http://help.github.com/>`_.
+ * the `NumPy's documentation <http://docs.scipy.org/doc/numpy/dev/index.html>`_.
+ * Matthew Brett's `Pydagogue <http://matthew-brett.github.com/pydagogue/>`_.
+
+Getting Started with Git
+------------------------
+
+`GitHub has instructions <http://help.github.com/set-up-git-redirect>`__ for installing git, 
+setting up your SSH key, and configuring git.  All these steps need to be completed before
+working seamlessly with your local repository and GitHub.
+
+Forking
+-------
+
+You will need your own fork to work on the code. Go to the `pandas project
+page <https://github.com/pydata/pandas>`_ and hit the *fork* button. You will
+want to clone your fork to your machine: ::
+
+    git clone git@github.com:your-user-name/pandas.git pandas-yourname
+    cd pandas-yourname
+    git remote add upstream git://github.com/pydata/pandas.git
+
+This creates the directory `pandas-yourname` and connects your repository to
+the upstream (main project) *pandas* repository.
+
+You will also need to hook up Travis-CI to your GitHub repository so the suite 
```

Comment:
It is indeed not needed anymore. But, if you want to get feedback from travis on a certain branch you push before you make it into a PR, then it can be useful. So maybe just a bit rephrasing that it is optional.